plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSessionProperties.java

```diff
@@ -387,6 +394,32 @@ public HiveSessionProperties(
                         "Require filter on partition column",
                         hiveConfig.isQueryPartitionFilterRequired(),
                         false),
+                new PropertyMetadata<>(
+                        QUERY_PARTITION_FILTER_REQUIRED_SCHEMAS,
+                        "Schemas for which query partition filter check can be applied.",
+                        new ArrayType(VARCHAR),
+                        Set.class,
+                        ImmutableSet.of(),
+                        false,
+                        object -> {
+                            if (object == null) {
+                                return ImmutableSet.of();
+                            }
+                            return ((Collection<?>) object).stream()
+                                    .peek(property -> {
+                                        if (property == null) {
+                                            throw new TrinoException(INVALID_ANALYZE_PROPERTY, format("Invalid null value in %s property", QUERY_PARTITION_FILTER_REQUIRED_SCHEMAS));
+                                        }
+                                    })
+                                    .map(String.class::cast)
+                                    .peek(property -> {
+                                        if (property.isEmpty()) {
+                                            throw new TrinoException(INVALID_ANALYZE_PROPERTY, format("Invalid empty value in %s property", QUERY_PARTITION_FILTER_REQUIRED_SCHEMAS));
+                                        }
+                                    })
```

Comment:
how to disable that if cannot specify neither null nor empty? I would allow specifying any of those and set value to empty list in either case.

Reply:
oh never mind - these are validators for elements.